common/utils/kubernetes.go

```diff
@@ -2,17 +2,24 @@ package utils
 
 import (
 	"encoding/json"
-
 	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"gopkg.in/yaml.v3"
 )
 
 func KubeconfigToJSON(kubeConfig pulumi.Output) pulumi.StringInput {
-	return kubeConfig.ApplyT(func(config interface{}) (string, error) {
-		jsonConfig, err := json.Marshal(config)
+	return kubeConfig.ApplyT(func(config string) (string, error) {
```

Comment:
This changes the signature of the `KubeconfigToJSON` function.
Won’t it break all the [callers](https://github.com/DataDog/test-infra-definitions/blob/c45b6947c391d5d70eac6fde60e1711b88925e0c/scenarios/aws/eks/run.go#L169) that are not updated in this PR ?

Reply:
I think the function name wasn't changed. The only change should be that we unmarshal as yaml first.

Reply:
Your change assumes that `kubeConfig` is a YAML string.
In this case, you should change the signature to take in a ` kubeConfigYAML pulumi.StringOutput` and perhaps change the name of the function to `KubeconfigYAMLToJSON` as it does not do exactly what it says.

Reply:
I did it in the last commit